### PR TITLE
[AR NumberRange] Coverage Improvements, Extractor and Spec Update

### PR DESCRIPTION
**Pull request by neudurgeshp**
**PR Includes**
- Number Definitions

- Specs
   - NumberModel
   - NumberRangeModel

- Number Model Extractor
   - Total Cases 279: Extractor Passed - 138, Full Passed - 37, Skipped - 141
   - New Cases 135: Extractor Passed - 130, Full Passed - 29, Skipped - 5
   - Integer 48: Extraction Passed - 44, Full Passed - 7, Skipped - 4
   - Fraction 56: Extraction Passed - 55, Full Passed - 12, Skipped - 1
   - Decimal 9: Extraction Passed - 9
   - Power 11: Extraction Passed - 11, Full Passed - 0, Skipped - 0
   - Negative 11: Extraction Passed - 11, Full Passed - 11

- Number Range Model Extractor

   - Total Test Cases 145: Extractor Passed - 105, Full Passed - 48, Skipped - 40
   - New Cases 78: Extractor Passed - 64, Full Passed - 12, Skipped- 14
   - Existing Cases 67: Extractor Passed - 41, Full Passed - 36, Skipped - 26

** One input got corrected by Language Expert.
"على الأكثر ١٠٠ وعلى الأقل ٢٠"

### CODE REVIEW

File: Specs/Number/Arabic/NumberRangeModel.json [hunk 1]
```diff
@@ -93,7 +93,7 @@
         "Text": "أكبر من ١٠٠ وأصغر من ٣٠٠",
         "TypeName": "numberrange",
         "Resolution": {
-          "value": "-100,300"
+          "value": "(100,300)"
```
Comment (tellarin) [inline Specs/Number/Arabic/NumberRangeModel.json:96]: This sentence was not with -100? Do we have one with it to add too?

File: Specs/Number/Arabic/NumberRangeModel.json [hunk 2]
```diff
@@ -149,18 +148,18 @@
     ]
   },
   {
-    "Input": "على الأكثر ١٠٠ وعلى الأقل ٢٠",
+    "Input": "الرقم ما بين ٢٠ و١٠٠.",
```
Comment (tellarin) [inline Specs/Number/Arabic/NumberRangeModel.json:151]: Please always add a comment in the PR description for such spec edits.

File: Specs/Number/Arabic/NumberRangeModel.json [hunk 3]
```diff
@@ -189,7 +188,7 @@
         "Text": "فوق ١٠٠٠ وتحت ١٥٠٠",
         "TypeName": "numberrange",
         "Resolution": {
-          "value": "-10,001,500"
```
Comment (tellarin) [inline Specs/Number/Arabic/NumberRangeModel.json:192]: Same, can we add another spec case that outputs a range with negative?

Comment (neudurgeshp) [reply Specs/Number/Arabic/NumberRangeModel.json:192]: Sure! will add into next iteration.

File: Specs/Number/Arabic/NumberRangeModel.json (hunk 1 above)
Comment (neudurgeshp) [reply Specs/Number/Arabic/NumberRangeModel.json:96]: Sure! will add into next iteration.

File: Specs/Number/Arabic/NumberRangeModel.json (hunk 2 above)
Comment (neudurgeshp) [reply Specs/Number/Arabic/NumberRangeModel.json:151]: Added in PR Description.

**Pull request merged**